### PR TITLE
Feature/staging uuid

### PR DESCRIPTION
**Pull request by silverdaz**
Staging area in now a flat folder, and filenames are UUIDs.

We do not need to clean any folders there. That solves a race.

### CODE REVIEW

File: lega/utils.py [hunk 1]
```diff
@@ -37,6 +38,21 @@ def only_central_ega(async_func):
...
+def check_error(func):
+    '''Decorator to store the raised exception in the database'''
+    @functools.wraps(func)
+    def wrapper(data):
+        try:
+            res = func(data)
+            res.__name__ = getattr(func, '__name__', None)
+            res.__qualname__ = getattr(func, '__qualname__', None)
+            return res
+        except Exception as e:
```
Comment (viklund) [inline lega/utils.py]: Just a thought, maybe you should catch LEGAException here. And have all our exceptions inherit from that class, that way you don't have to reraise AssertionErrors for example. Also you could do the LOG.warn in this one as well (but maybe you'd get the wrong package then).

File: lega/vault.py [hunk 2]
```diff
@@ -56,11 +56,7 @@ def work(data):
     db.finalize_file(file_id, starget, target.stat().st_size)
 
     # Send message to Archived queue
-    return {
-        'file_id': file_id,
-        'staging_folder': str(filepath.parent),
-        #'vault_name': starget,
-    }
+    return { 'file_id': file_id } # I could have the details in here. Fetching from DB instead.
```
Comment (viklund) [inline lega/vault.py]: This is for the future. What happens if `db.finalize_file` or `shutil.move` fails?

File: lega/verify.py [hunk 3]
```diff
@@ -22,39 +22,25 @@
...
-        if isinstance(e,AssertionError):
-            raise e
-        errmsg = f'{e.__class__.__name__}: {e!s}'
-        LOG.error(errmsg)
-        db.set_error(file_id, e)
-        
+    return {
+        'vault_name': vault_filename,
+        'org_name': filename
+    }
```
Comment (viklund) [inline lega/verify.py]: Nice! The `@check_error` made this very simple.

File: lega/worker.py [hunk 4]
```diff
@@ -79,66 +82,53 @@ def work(data):
     LOG.debug(f'Created id {file_id} for {data["filename"]}')
     assert file_id is not None, 'Ouch...database problem!'
 
+    # Check if file is in inbox
     inbox_filepath = inbox / filename
```
Comment (viklund) [inline lega/worker.py]: I realise that this is the supported method of using the Path module. But personally I dislike overloading infix math operators for this kind of thing. I had to think for a while before I realised that it wasn't division, personally I would prefer the following syntax:

    inbox_filepath = inbox.joinpath(filename)

But you can decide on this.

File: lega/worker.py [hunk 5]
```diff
@@ -79,66 +82,53 @@ def work(data):
...
+            errmsg = f"Invalid {hash_algo} checksum for {inbox_filepath}"
+            LOG.warning(errmsg)
+            raise exceptions.Checksum(hash_algo, f'for {inbox_filepath}')
+    LOG.debug(f'Valid {hash_algo} checksum for {inbox_filepath}')
+
+    # Create a unique name for the staging area
+    #unique_name = str(uuid.uuid4())
+    unique_name = str(uuid.uuid5(uuid.NAMESPACE_OID, 'lega'))
+    LOG.debug(f'Created an unique filename in the staging area: {unique_name}')
+    staging_filepath = staging_area / unique_name
```
Comment (viklund) [inline lega/worker.py]: I stared as this as well for a while before I realised what was going on. I do not understand why _Python_ of all languages endorses this kind of thing.

File: tools/start_swagger.sh [hunk 6]
```diff
@@ -1,9 +0,0 @@
-#!/usr/bin/env bash
-
-CONTAINER=ingestion-swagger
-
-# Kill the previous container
-docker kill ${CONTAINER} || true #&& docker rm  ${CONTAINER}
-
-# Starting RabbitMQ with docker
-docker run -it --rm -d --hostname localhost -p 8080:8080 --name ${CONTAINER} swaggerapi/swagger-ui
```
Comment (viklund) [inline tools/start_swagger.sh]: This should not be in this commit.

File: lega/verify.py [hunk 7]
```diff
@@ -33,28 +33,18 @@ def work(data):
...
-            os.rmdir(staging_folder) # raise exception is not empty
-            LOG.debug(f'Removing {staging_folder}')
-        except OSError:
-            pass
-
         return {
             'vault_name': vault_filename,
             'org_name': filename
         }
     except Exception as e:
```
Comment (viklund) [inline lega/verify.py]: You could catch just `LEGAException` here and then you woudn't need to do the `AssertionError` dance.

File: lega/worker.py [hunk 8]
```diff
@@ -130,14 +126,10 @@ def work(data):
         }
         LOG.debug(f"Reply message: {reply!r}")
         return reply
-
+
```
Comment (viklund) [inline lega/worker.py]: Nitpicky again: Maybe remove this whitespace?

File: lega/vault.py (hunk 2 above)
Comment (silverdaz) [inline lega/vault.py]: They now are also guarded with the check_error decorator

File: lega/verify.py (hunk 7 above)
Comment (silverdaz) [inline lega/verify.py]: Indeed.

File: lega/worker.py (hunk 4 above)
Comment (silverdaz) [inline lega/worker.py]: Nja... well... not a big deal to me...

File: tools/start_swagger.sh (hunk 6 above)
Comment (silverdaz) [inline tools/start_swagger.sh]: true

**Comment by silverdaz** [comment]
I have now re-included the check_error decorator. The file_id is inserted in the beginning when the message is brought from CentralEGA to the local broker.

If we add an endpoint into our frontend, we should remember to add the file_id in the message.

File: lega/utils.py (hunk 1 above)
Comment (viklund) [inline lega/utils.py]: I think you should delete commented lines in a commit, not keep them.

File: lega/utils.py [hunk 9]
```diff
@@ -42,23 +42,22 @@ def check_error(func):
...
         except Exception as e:
             if isinstance(e,AssertionError):
                 raise e
-            db.add_error(e)
+            db.set_error(file_id, e)
+            # frm = inspect.trace()[-1]
+            # mod = inspect.getmodule(frm[0])
+            # modname = mod.__name__ if mod else frm[1]
+            # print('Thrown from', modname)
     return wrapper
```
Comment (viklund) [inline lega/utils.py]: Same here, just remove it.

File: lega/amqp.py [hunk 10]
```diff
@@ -81,7 +81,7 @@ def process_request(channel, method_frame, props, body):
     from_channel.basic_consume(process_request, queue=from_queue)
     from_channel.start_consuming()
 
-def forward(from_channel, from_queue, to_channel, to_exchange, to_routing):
+def forward(from_channel, from_queue, to_channel, to_exchange, to_routing, transform=None):
     '''Blocking function, registering callback to be called, on each message from the queue `from_queue`
```
Comment (silverdaz) [inline lega/amqp.py]: ok, no pb

**Pull request merged**